internal/host/host.go

```diff
@@ -515,6 +517,58 @@ func (m *Manager) UpdateNTP(ctx context.Context, h *models.Host, ntpSources []*m
 	return db.Model(h).Update("ntp_sources", string(bytes)).Error
 }
 
+func (m *Manager) UpdateImageStatus(ctx context.Context, h *models.Host, newImageStatus *models.ContainerImageAvailability, db *gorm.DB) error {
+	var hostImageStatuses []*models.ContainerImageAvailability
+	imageFound := false
+
+	if h.ImagesStatus != "" {
+		err := json.Unmarshal([]byte(h.ImagesStatus), &hostImageStatuses)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to unmarshal image statuses for host %s", h.ID.String())
+		}
+	}
+
+	// Check if the image status already exist
+	for index := range hostImageStatuses {
+		if hostImageStatuses[index].Name != newImageStatus.Name {
+			continue
+		}
+
+		// Same result - Nothing to update
+		if hostImageStatuses[index].Result == newImageStatus.Result {
+			return nil
+		}
+
+		// New result - Get the bigger measurments. In case of a failure the measurments should be zero.
+		newImageStatus.SizeBytes = math.Max(hostImageStatuses[index].SizeBytes, newImageStatus.SizeBytes)
+		newImageStatus.Time = math.Max(hostImageStatuses[index].Time, newImageStatus.Time)
+		newImageStatus.DownloadRate = math.Max(hostImageStatuses[index].DownloadRate, newImageStatus.DownloadRate)
```

Comment:
Do we want to update if it's not a new image?

Reply:
I think that the DB should represent the status of the agent and hence it is still being updated.
We decided not to look at it on the events/metrics level (only new image status response).

Reply:
Fair enough

Reply:
But wait, this isn't representing the agent status this is taking the max, meaning you will save to DB the max value and not the new value.

Reply:
That's right. The last valid value (in case of a failure, the statistics will remain)